src/libs/common/ide/Location.ts

```diff
@@ -0,0 +1,33 @@
+import { URI } from "vscode-uri";
+import Position from "./Position";
+import Range from "./Range";
+
+/**
+ * Represents a location inside a resource, such as a line
+ * inside a text file.
+ */
+export default class Location {
```

Comment:
I wonder if we want a version of this that has a document pointer. Also where do you need this one?

Reply:
It's used by Tree sitter getNodeAtLocation